src/append/mod.rs

```diff
@@ -23,16 +23,43 @@ pub mod rolling_file;
 
 #[cfg(any(feature = "file_appender", feature = "rolling_file_appender"))]
 mod env_util {
-    #[allow(clippy::redundant_clone)]
+    const ENV_VAR_PREFIX: &str = "$ENV{";
+    const ENV_VAR_SUFFIX: char = '}';
+
     pub fn expand_env_vars(path: std::path::PathBuf) -> std::path::PathBuf {
-        let mut path: String = path.to_string_lossy().into();
-        let matcher = regex::Regex::new(r#"\$ENV\{([\w][\w|\d|\.|_]*)\}"#).unwrap();
-        matcher.captures_iter(&path.clone()).for_each(|c| {
-            if let Ok(s) = std::env::var(&c[1]) {
-                path = path.replace(&c[0], &s);
+        let path: String = path.to_string_lossy().into();
+        let mut outpath: String = path.clone();
```

Comment:
You could build this incrementally to avoid many copies if there are many env var replacements. Not a requirement, just an idea.

Reply:
Yes, I left a comment for line 58. Though it feels bad to use a hand-written scanner, but then do this at the end, I don't think it's critical. Let me think about this a bit more. Maybe it's easier than I think.